tensorboard/webapp/metrics/effects/metrics_effects_test.ts

```diff
@@ -510,6 +546,43 @@ describe('metrics effects', () => {
 
         expect(fetchTimeSeriesSpy).toHaveBeenCalledTimes(2);
       });
+
+      it('does not send requests to experiments lacking a cards tag', () => {
+        store.overrideSelector(getActivePlugin, METRICS_PLUGIN_ID);
+        store.overrideSelector(selectors.getExperimentIdsFromRoute, [
+          'exp1',
+          'exp2',
+        ]);
+        store.overrideSelector(
+          selectors.getVisibleCardIdSet,
+          new Set(['card1', 'card2'])
+        );
+        provideCardFetchInfo([
+          {id: 'card1', tag: 'tagA'},
+          {id: 'card2', tag: 'tagB'},
+        ]);
+        store.refreshState();
+
+        const effectFetchTimeSeriesSpy = spyOn(
+          effects as any,
+          'fetchTimeSeries'
+        ).and.stub();
+
+        actions$.next(coreActions.manualReload());
+
+        expect(effectFetchTimeSeriesSpy).toHaveBeenCalledTimes(2);
+        expect(effectFetchTimeSeriesSpy).toHaveBeenCalledWith({
+          plugin: 'scalars',
+          tag: 'tagA',
+          experimentIds: ['exp1'],
+        });
+
+        expect(effectFetchTimeSeriesSpy).toHaveBeenCalledWith({
+          plugin: 'scalars',
```

Comment:
Given the complexity of the logic you've written it would be good to see more rigourous testing. A single test case doesn't really seem to cut it.

A couple that pop up in my head:
* A test case where some image requests make it through the new filter.
* A test case where some histogram requests make it through the new filter.